enclave/server.ts

```diff
@@ -51,11 +51,57 @@ const nStates = new ethers.Contract(
     signer
 );
 
+type ClaimedMoved = {
+    socketId: string;
+    uFrom: Tile;
+    uTo: Tile;
+    hUFrom: string;
+    hUTo: string;
+};
+
 /*
  * Enclave's internal belief on game state stored in Board object.
  */
 let b: Board;
 
+/*
+ *
+ */
+let players: Player[] = [];
+
+/*
+ * List of claimed moves, pend for contract to emit event.
+ */
+let claimedMoves: ClaimedMoved[] = [];
+
+/*
+ * Dev function for spawning a player on the map. Returns true if player is
+ * spawned in. Return false otherwise.
+ */
+async function spawn(
+    socket: Socket,
+    symbol: string,
+    l: Location,
+    pubkey: string
+) {
+    // If player already exists, update the socketId. Otherwise, instantiate.
+    let alreadySpawned = false;
+    for (let i = 0; i < players.length; i++) {
```

Comment:
could we make this a map instead of looping through, or is there a specific reason you wanted a player list

Reply:
maybe players is a map from socket id to Player, or the other way around. also not sure what you mean by the second comment here